lib/europeana/api/record/hierarchy.rb

```diff
@@ -0,0 +1,83 @@
+require 'rubytree'
+require 'active_support/core_ext/object/blank'
+
+module Europeana
+  module API
+    class Record
+      ##
+      # Retrieve record hierarchies over the Europeana API
+      class Hierarchy < Tree::TreeNode
+        include Requestable
+
+        attr_reader :record
+
+        ##
+        # @param record [Europeana::API::Record]
+        def initialize(record)
+          @record = record
+          super(record.id, execute_request[:self])
+          fetch_family
+        end
+
+        ##
+        # Retrieves parent, children and siblings of this node
+        def fetch_family
```

Comment:
Assignment Branch Condition size for fetch_family is too high. [28.23/15]<br>Method has too many lines. [17/10]